src/Analyzer/CodeAnalyzer.cs

```diff
@@ -152,29 +169,10 @@ public async Task Analyze(string file, CancellationToken cancellationToken)
 
         public ImmutableArray<DiagnosticAnalyzer> GetAnalyzers()
         {
-            if (PatternIds is null)
+            if(PatternIds is null)
             {
-                if (File.Exists(sonarConfigurationPath))
-                {
-                    var xmlDoc = XDocument.Load(sonarConfigurationPath);
-                    var analysisInput = xmlDoc.Element("AnalysisInput");
-                    var rules = analysisInput.Element("Rules"); 
-                    if (rules is null)
-                    {
-                        return availableAnalyzers;
-                    }
-                    else
-                    {
-                        PatternIds = rules.Elements("Rule").Select(e => e.Elements("Key").Single().Value)
-                            .ToImmutableList();
-                    }
-                }
-                else
-                {
-                    return availableAnalyzers;
-                }
+                return availableAnalyzers;
             }
```

Comment:
So you are initializing them before in the constructor instead of creating them in the getter, right?

Reply:
Yes. Exactly. This way we can use the variable without having to worry if we already fetched the analysers